### PR TITLE
SDK - Refactored _func_to_component_spec to split code generation from signature analysis

### PR DESCRIPTION
**Pull request by Ark-kun**
This refactoring is needed to allow using different code generators while sharing the same function signature analysis code.

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1334)

### CODE REVIEW

File: sdk/python/kfp/components/_python_op.py [hunk 1]
```diff
@@ -45,39 +46,13 @@ def _python_function_name_to_component_name(name):
...
-            base_image = decorator_base_image
-    else:
-        if base_image is None:
-            raise ValueError('base_image cannot be None')
-
-    import inspect
-    import re
-    from collections import OrderedDict
-    
+def _create_component_interface_spec_from_function_signature(func) -> ComponentSpec:
```
Comment (gaoning777) [inline sdk/python/kfp/components/_python_op.py]: Why not _create_component_spec_from_function_signature?

File: sdk/python/kfp/components/_python_op.py [hunk 2]
```diff
@@ -113,24 +86,61 @@ def annotation_to_type_struct(annotation):
                 type=type_struct,
             )
             outputs.append(output_spec)
-            extra_output_names.append(field_name)
-            arguments.append(OutputPathPlaceholder(field_name))
     elif signature.return_annotation is not None and signature.return_annotation != inspect.Parameter.empty:
```
Comment (gaoning777) [inline sdk/python/kfp/components/_python_op.py]: What if it is a tuple, should each output be extracted? This looks like it only extract multiple outputs when it is a NamedTuple.

Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: How does the tuple output _annotation_ look like?
We need to know the output signature - the names of the outputs (and their types).

File: sdk/python/kfp/components/_python_op.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: >How about just _extract_component()?

Sounds good. I'd like to have the `interface` word there though.
How about `_extract_component_interface`?

Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: Done

File: sdk/python/kfp/components/_python_op.py (hunk 2 above)
Comment (gaoning777) [reply sdk/python/kfp/components/_python_op.py]: What if it is a dict? e.g. https://github.com/kubeflow/pipelines/blob/7e29b431bc7696b5c5468dc6d714d18858d63ecd/sdk/python/tests/dsl/component_tests.py#L33
I think this function can later be used in the dsl as well.

Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: Oh. I see. I agree that we should improve the type support here. I just want to keep refactorings separate from changes. It's much easier to track and review thing that way.
I'll expand this function in another PR.

**Pull request merged**